### PR TITLE
Add OrdinalPageIndex and EncodingInfo

### PR DESCRIPTION
**Pull request by imay**

### CODE REVIEW

File: be/src/olap/rowset/segment_v2/encoding_info.h [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+namespace doris {
+
+class TypeInfo;
+
+namespace segment_v2 {
+
+class PageBuilder;
+class PageDecoder;
+
+
```
Comment (kangpinghuang) [inline be/src/olap/rowset/segment_v2/encoding_info.h]: additional empty line

File: be/src/olap/rowset/segment_v2/encoding_info.cpp [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+};
+
+EncodingInfoResolver::EncodingInfoResolver() {
+    _add_map<OLAP_FIELD_TYPE_TINYINT, PLAIN_ENCODING>();
+    _add_map<OLAP_FIELD_TYPE_SMALLINT, PLAIN_ENCODING>();
+    _add_map<OLAP_FIELD_TYPE_INT, PLAIN_ENCODING>();
+    _add_map<OLAP_FIELD_TYPE_BIGINT, PLAIN_ENCODING>();
+    _add_map<OLAP_FIELD_TYPE_LARGEINT, PLAIN_ENCODING>();
+    _add_map<OLAP_FIELD_TYPE_FLOAT, PLAIN_ENCODING>();
+    _add_map<OLAP_FIELD_TYPE_DOUBLE, PLAIN_ENCODING>();
```
Comment (kangpinghuang) [inline be/src/olap/rowset/segment_v2/encoding_info.cpp]: Other encoding type will be added in the future?

File: be/src/olap/rowset/segment_v2/ordinal_page_index.h [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+    void append_entry(rowid_t rid, const PagePointer& page) {
+        // rid
+        put_varint32(&_buffer, rid);
+        // page pointer
+        page.encode_to(&_buffer);
+        _num_pages++;
+    }
+
+    Slice finish() {
+        encode_fixed32_le((uint8_t*)_buffer.data(), _num_pages);
```
Comment (kangpinghuang) [inline be/src/olap/rowset/segment_v2/ordinal_page_index.h]: add comment here. 
eg: add _num_pages to the first 4 bytes

File: be/src/olap/rowset/segment_v2/ordinal_page_index.h [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+// Content:
+//      array of index_pair
+// index_pair:
+//      Ordinal (4 Bytes)
+//      PagePointer (8 Bytes)
+class OrdinalPageIndexBuilder {
+public:
+    OrdinalPageIndexBuilder() : _num_pages(0) {
+        _buffer.reserve(4 * 1024);
+        _buffer.resize(4);
```
Comment (kangpinghuang) [inline be/src/olap/rowset/segment_v2/ordinal_page_index.h]: add a comment, eg: reserve the first 4 bytes for number of elements

File: be/src/olap/rowset/segment_v2/ordinal_page_index.cpp [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+Status OrdinalPageIndex::load() {
+    DCHECK_GE(_data.size, _header_size()) << "block size must greate than header";
+    const uint8_t* ptr = (const uint8_t*)_data.data;
+    const uint8_t* limit = (const uint8_t*)_data.data + _data.size;
+
+    _num_pages = decode_fixed32_le(ptr);
+    ptr += 4;
+
+    _rowids = new rowid_t[_num_pages];
+    _pages = new PagePointer[_num_pages];
```
Comment (kangpinghuang) [inline be/src/olap/rowset/segment_v2/ordinal_page_index.cpp]: I think it better to add std::nothrow and check whether the pointer is nullptr

**Review by kangpinghuang — CHANGES_REQUESTED** [review]

File: be/src/olap/rowset/segment_v2/encoding_info.cpp (hunk 2 above)
Comment (imay) [reply be/src/olap/rowset/segment_v2/encoding_info.cpp]: Yes, this is temporary

File: be/src/olap/rowset/segment_v2/ordinal_page_index.cpp (hunk 5 above)
Comment (imay) [reply be/src/olap/rowset/segment_v2/ordinal_page_index.cpp]: there is no need to do this, just crash the process

File: be/src/olap/rowset/segment_v2/ordinal_page_index.h (hunk 4 above)
Comment (imay) [reply be/src/olap/rowset/segment_v2/ordinal_page_index.h]: OK, I will fixed

File: be/src/olap/rowset/segment_v2/encoding_info.h (hunk 1 above)
Comment (imay) [reply be/src/olap/rowset/segment_v2/encoding_info.h]: OK

File: be/src/olap/rowset/segment_v2/page_pointer.h [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+// specific language governing permissions and limitations
+// under the License.
+
+#pragma once
+
+#include <cstdint>
+
+#include "util/coding.h"
+
+
```
[the remark this answers is not in the document]
Comment (imay) [reply be/src/olap/rowset/segment_v2/page_pointer.h]: OK

**Review by kangpinghuang — APPROVED** [review]

**Review by chaoyli — APPROVED** [review]

**Pull request merged**